src/installer/src/main.rs

```diff
@@ -0,0 +1,232 @@
+        locations.push(path.to_path_buf());
+    }
+    if let Some(path) = user_dirs.desktop_dir() {
+        locations.push(path.to_path_buf());
+    }
+    if let Some(path) = user_dirs.download_dir() {
+        locations.push(path.to_path_buf());
+    }
+    if let Some(path) = user_dirs.document_dir() {
+        locations.push(path.to_path_buf());
+    }
+    if let Some(path) = user_dirs.public_dir() {
+        locations.push(path.to_path_buf());
+    }
+
+    let candidate = Path::new("/opt");
+    if candidate.exists() {
+        locations.push(candidate.to_path_buf());
+    }
+    let candidate = Path::new("/usr");
+    if candidate.exists() {
+        locations.push(candidate.to_path_buf());
+    }
+
+    locations
+}
+
+/// Returns None if Ghidra were not found. Else returns path to Ghidra, which might be user selected.
+fn find_ghidra() -> Result<PathBuf> {
+    let mut ghidra_locations: Vec<PathBuf> = get_search_locations()
+        .into_iter()
+        .filter_map(|x| search_for_ghidrarun(&x))
+        .collect();
+
+    match ghidra_locations.len() {
+        0 => Err(Error::new(ErrorKind::NotFound, "Ghidra not found").into()),
+        1 => Ok(ghidra_locations.pop().unwrap()),
+        _ => select_ghidra_version(ghidra_locations),
+    }
+}
+
+/// Searches for a file containing "ghidraRun" at provided path recursively.
+fn search_for_ghidrarun(entry_path: &Path) -> Option<PathBuf> {
+    for entry in WalkDir::new(entry_path)
+        .into_iter()
+        .filter_map(|e| e.ok())
+        .filter(|e| e.metadata().unwrap().is_file())
+    {
+        if entry.file_name().to_str().unwrap().contains("ghidraRun") {
+            let mut hit = entry.into_path();
+            hit.pop();
+            return Some(hit);
+        }
+    }
+    None
+}
+
+/// Determines Ghidra versions and provides selection interface for the user.
+fn select_ghidra_version(mut ghidra_locations: Vec<PathBuf>) -> Result<PathBuf> {
+    let mut i = 0;
+    while i < ghidra_locations.len() {
+        ghidra_locations[i].push("Ghidra/application.properties");
+        let app_prop = std::fs::read_to_string(&ghidra_locations[i]);
+
+        match app_prop {
+            Ok(properties) => {
+                let version: &str = properties
+                    .lines()
+                    .filter_map(|x| x.strip_prefix("application.version="))
+                    .next()
+                    .unwrap_or("?");
+                ghidra_locations[i].pop();
+                ghidra_locations[i].pop();
+                println!(
+                    "Use Ghidra at: {} [v{}]? ({})",
+                    ghidra_locations[i].display(),
+                    version,
+                    i
+                );
+                i += 1;
+            }
+
+            Err(_) => {
+                ghidra_locations.remove(i);
+            }
+        }
+    }
+    if ghidra_locations.is_empty() {
+        return Err(Error::new(ErrorKind::NotFound, "Ghidra not found").into());
```

Comment:
Since we are using the anyhow crate, we can write this more concise as
```rust
return Err(anyhow!("Ghidra not found."));
```
This is true for all error messages you use.